include/nccl_ofi_ep_addr_list.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef NCCL_OFI_EP_ADDR_LIST_H
+#define NCCL_OFI_EP_ADDR_LIST_H
+
+#include "nccl_ofi.h"
+
+struct nccl_ofi_ep_addr_list;
+typedef struct nccl_ofi_ep_addr_list nccl_ofi_ep_addr_list_t;
+
+/**
+ * Initialize an endpoint-address-set pair list
```

Comment:
who allocates memory?

Reply:
This function does. Will note in docstring.